drivers/overlay/mount.go

```diff
@@ -141,14 +141,27 @@ func mountOverlayFromMain() {
 	// the new value for the list of lowers, because it's shorter.
 	if lowerv != "" {
 		lowers := strings.Split(lowerv, ":")
-		for i := range lowers {
-			lowerFd, err := unix.Open(lowers[i], unix.O_RDONLY, 0)
+		var newLowers []string
+		dataOnly := false
+		for _, lowerPath := range lowers {
+			if lowerPath == "" {
+				dataOnly = true
```

Comment:
Should this loop be enforcing the "once we see a data-only layer, they should all be data-only layers" requirement that the kernel has?

Reply:
not sure, if the input is not correct we'd get a similar error from `Mount` as we do if we call directly the syscall and we are not doing any validation in the `mountOverlayFromMain` function.

I don't think we need it, but if you prefer to have it, I'll add it